### DISCUSSION TITLE
Graph does not scroll to the left

### DISCUSSION THREAD

**Discussion by mmMOLINARI** [grafana/grafana-json-datasource#92] [Q&A] [answered]
Hi all,
I just started using grafana and the JSON API plug-in to plot data retrieved from several network devices. None of the API responses come with a time stamp, so I created a shim layer to add the timestamp to the response. My timestamp is epoch/unix in seconds, and my counter is an integer.
All looks good when I do an inspect query: the counter shows the expected value, and the timestamp increments every 5 seconds, which is the interval I set.
I can display the counter and timestamp using a Table and it shows correctly. However, when I used a Graph, I only see a vertical line (or a dot, depending on the Graph settings) stuck at the right (the height matches the counters), while the X axis increments as expected. Meaning, the graph does not roll to the left. I included some relevant images in this message.
Am I missing something? 
Thanks in advance,
Marcelo

![image](https://user-images.githubusercontent.com/68390677/114491615-c161a380-9bcb-11eb-9222-3d56799a9138.png)
![image](https://user-images.githubusercontent.com/68390677/114491770-071e6c00-9bcc-11eb-9125-527d897249e8.png)
![image](https://user-images.githubusercontent.com/68390677/114491241-27016000-9bcb-11eb-98e6-edebdfbd9a74.png)
![image](https://user-images.githubusercontent.com/68390677/114491857-37660a80-9bcc-11eb-9231-abd307903028.png)

**Comment by marcusolsson** [comment, marked as answer]
It seems like you want the plugin to store previous values of the total count. This is not possible as it would require the plugin to maintain state. Currently, the endpoint needs to return the entire time series. 

If the plugin were to store previous values, they would be stored as transient state. Whenever you refresh the dashboard, all previous values would be wiped.

**Reply by mmMOLINARI** [reply]
Hi Marcus,
I added more data points to the API response. It now includes 7 items in 5 seconds interval. The query response is shown below:
![image](https://user-images.githubusercontent.com/68390677/114657791-9a77a000-9ca5-11eb-929f-7044d3861209.png)
The timestamps increment with every query.
I can see a table with all values, and the timestamps also increment:
![image](https://user-images.githubusercontent.com/68390677/114657718-82078580-9ca5-11eb-8e5a-901fe3844725.png)
However, my graph still does not scroll left:
![image](https://user-images.githubusercontent.com/68390677/114657923-e0ccff00-9ca5-11eb-9d20-b92aa9e7d228.png)
Shouldn't it at least show my 7 data points now?

Thanks again!

**Reply by mmMOLINARI** [reply]
Got it, my simulated data points were using timestamps in the future, to the right of X axis. I had added time intervals to the original time instead of subtracting. I fixed the data points and now I can see the graph:

![image](https://user-images.githubusercontent.com/68390677/114660149-ad8c6f00-9ca9-11eb-8a79-e96283075fdb.png)

Thanks again!

**Reply by marcusolsson** [reply]
Glad you got it working!

**Reply by marcusolsson** [reply]
I've added a [Troubleshooting](https://marcus.se.net/grafana-json-datasource/troubleshooting) page to the docs site to hopefully help future users with the same question.

**Comment by mmMOLINARI** [comment]
The link in your message it taking us to the wrong prometheus URL:
https://prometheus.org/



Marcelo Molinari
Sr. Solutions Architect, WW Solutions Engineering

COMMSCOPE
now meets next

Mobile: +1 954 9947539
Email: ***@***.******@***.***>





From: Marcus Olsson ***@***.***>
Reply-To: marcusolsson/grafana-json-datasource ***@***.***>
Date: Wednesday, April 14, 2021 at 1:28 AM
To: marcusolsson/grafana-json-datasource ***@***.***>
Cc: "Molinari, Marcelo" ***@***.***>, Author ***@***.***>
Subject: Re: [marcusolsson/grafana-json-datasource] Graph does not roll to the left (#92)

I've added a Troubleshooting page to the docs site to hopefully help future users with the same question. —You are receiving this because you authored the thread.Reply to this email directly, view it
External ***@***.******@***.***>)
  Report This Email<https://shared.outlook.inky.com/report?id=Y29tbXNjb3BlL21hcmNlbG8ubW9saW5hcmlAY29tbXNjb3BlLmNvbS9hMjZhMDRmZjExZGZiNzVlMzY3NWVhM2FmNmY5ODZhOC8xNjE4Mzg4ODgxLjc4#key=5c8a948e5c6f619ed839b7c625f5645b>  FAQ<https://www.inky.com/banner-faq/>  Protection by INKY<https://www.inky.com>


I've added a Troubleshooting<https://secure-web.cisco.com/1gH0gUMR7mfbrjMavWq6Cbz72NjI0sZcWW70CFJHN8EqfLhVnOB1RlXzpo5VtxZc2lpI5O8Bb74MfD8nc3s7QrEfwb8hMvK8Rj5ruEdfITnftUHlpj6T_Oy1_88Mq7PHNf67TDlwe7DpRWyH43a6unPbqr6oN6qv5_OrEK8S50gR5eWLQUqxypuuTjsF1tur7mCz74Wbjg9Gu9sciqMeiAloUDW-i2OS0Kl1UxNGIVv2y97wlukbHXfGpXhoq5UayuAJA6c-qHeoT_UeUDkXgZdPn4x7WJiQGwLa2cThv3lpXIxTEEgDNeo6liNCuUqB2/https%3A%2F%2Fmarcus.se.net%2Fgrafana-json-datasource%2Ftroubleshooting> page to the docs site to hopefully help future users with the same question.

**Reply by marcusolsson** [reply]
Gah, thanks for the catch! 🙏 I'll update it right away
